File: src/main/java/com/tjh/riskfactor/config/MethodSecurityConfig.java
package com.tjh.riskfactor.config;

import lombok.RequiredArgsConstructor;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
import org.springframework.security.config.annotation.method.configuration.GlobalMethodSecurityConfiguration;

import com.tjh.riskfactor.security.PermissionEvaluator;

@Configuration
@EnableGlobalMethodSecurity(
    prePostEnabled = true
)
@RequiredArgsConstructor
public class MethodSecurityConfig extends GlobalMethodSecurityConfiguration {

    private final PermissionEvaluator evaluator;

    @Bean(name = "e")
    public PermissionEvaluator permissionEvaluator() {
        return evaluator;
    }

}

File: backend/src/main/java/com/tjh/riskfactor/config/MethodSecurityConfig.kt
package com.tjh.riskfactor.config

import org.springframework.context.annotation.Bean
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Primary
import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity
import org.springframework.security.config.annotation.method.configuration.GlobalMethodSecurityConfiguration

import com.tjh.riskfactor.security.PermissionEvaluator

@Configuration
@EnableGlobalMethodSecurity(
    prePostEnabled = true
)
class MethodSecurityConfig: GlobalMethodSecurityConfiguration() {

    @Autowired lateinit var evaluator: PermissionEvaluator

    @Bean @Primary
    fun e() = evaluator

}
